src/utilities/index.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Utility to help with exhaustive checking of if-else if-else and switches,
+ * especially useful for reducers and components.
+ *
+ * @param type - type to ensure is exhaustively checked
+ */
+export function checkExhaustively(type: never) {
+    return type;
```

Comment:
Not sure how this function is supposed to work, but based on intuition it does not look fully implemented..?

Reply:
```ts
// Here's an enum!
enum Fruit {
  APPLE = "APPLE",
  PEAR = "PEAR",
  BANANA = "BANANA"
}

// Here's a function!
function eatFruit(fruit: Fruit) {
    // Ideally we'd like there to be a case for all Fruit.
    // This is a piece of cake with only 3 types, but let's pretend we miss one.
    switch (fruit) {
        case Fruit.BANANA:
            console.log("Tasty banana!");
        case Fruit.PEAR:
            console.log("Tasty pear!");
        default:
            console.log("Not a fruit I know about.");
    }
    // ...there's no error, linting, compilation, or otherwise.
    // ...if a case is forgotten, it can take time and effort to
    // debug it, plus there could be dead code.
```
Enter the mighty checkExhaustively!
```ts
function eatFruit(fruit: Fruit) {
    // Ideally we'd like there to be a case for all Fruit.
    // This is a piece of cake with only 3 types, but let's pretend we miss one.
    switch (fruit) {
        case Fruit.BANANA:
            console.log("Tasty banana!");
        case Fruit.PEAR:
            console.log("Tasty pear!");
        default:
            checkExhaustively(fruit); // Linting error! Bad programmer! You forgot the apple!
            console.log("Not a fruit I know about.");
    }
```